src/createAnimatedComponent.tsx

```diff
@@ -580,16 +581,27 @@ export default function createAnimatedComponent(
 
     _filterNonAnimatedStyle(inputStyle: StyleProps) {
       const style: StyleProps = {};
+      let changed = false;
       for (const key in inputStyle) {
         const value = inputStyle[key];
         if (!hasAnimatedNodes(value)) {
           style[key] = value;
+          changed = changed || style[key] !== this._lastSentStyle[key];
         } else if (value instanceof AnimatedValue) {
           // if any style in animated component is set directly to the `Value` we set those styles to the first value of `Value` node in order
           // to avoid flash of default styles when `Value` is being asynchrounously sent via bridge and initialized in the native side.
           style[key] = value._startingValue;
+          changed = changed || style[key] !== this._lastSentStyle[key];
         }
       }
+      if (!changed) {
+        const lastKeys = Object.keys(this._lastSentStyle);
+        const inputKeys = Object.keys(inputStyle);
+        if (lastKeys.every((k) => inputKeys.includes(k))) {
```

Comment:
I would at least make it a for of loop with `key in inputStyle` so we won't have a loop (`includes`) for every loop iteration (`every`) for no reason.

Reply:
I added some code to avoid the double loop.
We have to iterate through the lastStyle, since we already know all relevant information from the input style is present in lastStyle. What we have to make sure is that no extra information is being sent through lastStyle (no extra key in lastStyle that is not present on inputStyle).

I could populate the Set in the previous for, but not sure what is faster (initializing the set directly with an array, or adding one by one the elements).